action.yml

```
@@ -38,6 +38,12 @@ runs:
         elif [ $(find "." -maxdepth 1 -name "pnpm-lock.yaml") ]; then
             echo "PACKAGE_MANAGER=pnpm" >> $GITHUB_ENV
             echo "LOCKFILE=pnpm-lock.yaml" >> $GITHUB_ENV
+            # If packageManager field is not present, use latest version.
+            if ! grep -q "\"packageManager\":"  package.json; then
```

```suggestion
            if ! jq -e '.packageManager' package.json > /dev/null; then
```

I'd guess a more thorough test (including depth of the field) would be using `jq`?

---

Ah I also considered it but wasn’t sure if `jq` is always available?

---

Oh, yeah, looks like it is: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#tools